### ISSUE TITLE
tlg0015.tlg001.opp-grc1.xml

### ISSUE THREAD

**Issue by sonofmun** [gcelano/POStaggedAncientGreekXML#1]
This text has 3 citation levels in the 1st1KGreek repo (https://github.com/OpenGreekAndLatin/First1KGreek/blob/master/data/tlg0015/tlg001/tlg0015.tlg001.opp-grc1.xml#L45-L47), but only 2 are showing up in the `@p` elements here. The deepest level comes from `<seg>` elements, so perhaps that is messing it up?

**Comment by sonofmun** [comment]
It looks like it might really be a problem with having `<seg>` as one of the citation levels. I checked another text with 3 citation levels (tlg2018.tlg001.opp-grc1.xml) and it shows up just fine.

**Comment by sonofmun** [comment]
Some more information here: all three citation levels show up correctly at http://cts.dh.uni-leipzig.de/read/greekLit/tlg0015/tlg001/opp-grc1

**Comment by gcelano** [comment]
Yes, if the level is in seg, that is a problem for my algorithm, which assumes div (or l). Is there a particular reason why we want seg instead of div?

**Comment by sonofmun** [comment]
I am not sure why we used seg here. But there are definitely texts in there that use `<p>` elements, because they simply number the paragraphs as they stand. Perhaps you could simple grab the xpath statements from /TEI/teiHeader/encodingDesc/refsDecl/cRefPattern? That way you wouldn't have to guess what the citation scheme might be. If a text is CapiTainS-compliant, then the information there should describe the citation scheme of the document correctly.

**Comment by gcelano** [comment]
I thought I could avoid to use RefPattern, which requires more computation in my script. While p having @n makes sense (I have to correct this in my script), still I would try to make "citation markup" as uniform as possible: seg with @type="noparse" is used in 

https://github.com/PerseusDL/canonical-greekLit/blob/master/data/tlg0525/tlg001/tlg0525.tlg001.perseus-grc2.xml

as an "editorial" element (whose meaning is not always clear to me). So, for the function of delimiting a text part, I would go for the more specialized div than seg.

**Comment by sonofmun** [comment]
The problem with `<seg>` makes sense. We will need to look at that text and try to figure out why we used it.
As far as parsing the cRefPattern, I would think that you really should do this instead of assuming a closed set of citation markers if you wanted to make your script more generalizable. Especially if you want to run it in the future on texts that do not come directly from us, I would think that parsing the cRefPattern for every text would be a fairly computationally inexpensive way to make sure you don't miss any pieces of text.

**Comment by gcelano** [comment]
I agree. I'll incorporate it.

**Comment by sonofmun** [comment]
Let me know if you need help testing or at some other point. I am happy to help.

**Comment by gcelano** [comment]
@sonofmun, is it always @n that contains the cts:urn (no matter what the element is)?

**Comment by sonofmun** [comment]
Yes, I am almost certain that this is true. `@n` on the `<div type="edition">` element has the edition level URN and the cRefPattern always returns the n attribute. It is different in the `__cts__.xml` metadata files. There the URNs are in `@urn, @workUrn,` and `@groupUrn` attributes. But I assume you are talking about the text files and not the metadata files.

**Comment by gcelano** [comment]
Thanks, this is helpful

**Comment by gcelano** [comment]
@sonofmun, I have re-assigned the cts-urn to the tokens here 

https://github.com/gcelano/CTSAncientGreekXML

According to my script, the elements relevant to cts structure are div, seg, l, and p. If you spot any error, please tell me.
